test/Npgsql.Tests/NotificationTests.cs

```diff
@@ -180,35 +181,53 @@ public void WaitAsync_cancellation()
         }
     }
 
-    [Test]
+    [Test, NonParallelizable]
     public void Wait_breaks_connection()
     {
-        using var _ = CreateTempPool(ConnectionString, out var connString);
-        using var conn = OpenConnection(connString);
-        Task.Delay(1000).ContinueWith(t =>
+        try
         {
-            using var conn2 = OpenConnection();
-            conn2.ExecuteNonQuery($"SELECT pg_terminate_backend({conn.ProcessID})");
-        });
+            using var _ = CreateTempPool(ConnectionString, out var connString);
+            using var conn = OpenConnection(connString);
+            Task.Delay(1000).ContinueWith(t =>
+            {
+                using var conn2 = OpenConnection();
+                conn2.ExecuteNonQuery($"SELECT pg_terminate_backend({conn.ProcessID})");
+            });
 
-        var pgEx = Assert.Throws<PostgresException>(conn.Wait)!;
-        Assert.That(pgEx.SqlState, Is.EqualTo(PostgresErrorCodes.AdminShutdown));
-        Assert.That(conn.FullState, Is.EqualTo(ConnectionState.Broken));
+            var pgEx = Assert.Throws<PostgresException>(conn.Wait)!;
+            Assert.That(pgEx.SqlState, Is.EqualTo(PostgresErrorCodes.AdminShutdown));
+            Assert.That(conn.FullState, Is.EqualTo(ConnectionState.Broken));
+        }
+        finally
+        {
+            // Since this breaks the connection with a critical failure, clear the cluster state cache to prevent multihost tests from
+            // failing.
+            ClusterStateCache.Clear();
+        }
     }
 
     [Test]
```

Comment:
This one should also be marked as `NonParallelizable` since we clear the cluster state cache.